pkg/util/docker/dockerfile/builder/client.go

```diff
@@ -134,6 +149,23 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 			},
 		}
 		if mustStart {
+			// Transient mounts only make sense on images that will be running processes
+			if len(e.TransientMounts) > 0 {
+				volumeName, err := randSeq(imageSafeCharacters, 24)
+				if err != nil {
+					return err
+				}
+				v, err := e.Client.CreateVolume(docker.CreateVolumeOptions{Name: volumeName})
+				if err != nil {
+					return err
+				}
+				defer e.cleanupVolume(volumeName)
+				sharedMount = v.Mountpoint
+				opts.HostConfig = &docker.HostConfig{
+					Binds: []string{sharedMount + ":/tmp/__temporarymount"},
+				}
```

Comment:
I want to do this without being local to the machine, so I need a volume.  If you try to bind mount something before it exists, it gets bind mounted as a dir.  I don't want the bind mounts to be writable (otherwise we have a quota problem), so I need a two phase mount + copy, then remount.